src/rust/engine/fs/src/lib.rs

```diff
@@ -541,20 +541,35 @@ pub struct GlobWithSource {
 }
 
 ///
-/// All Stats consumed or return by this type are relative to the root.
+/// All Stats consumed or returned by this type are relative to the root.
+///
+/// If `symlink_aware` is true (as it is by default), `scandir` will produce `Link` entries so that
+/// a consumer can explicitly track their expansion. Otherwise, it will allow the operating system
+/// to expand the link to its underlying type without regard to the links traversed, and `scandir`
+/// will produce only `Dir` and `File` entries.
 ///
 #[derive(Clone)]
 pub struct PosixFS {
   root: Dir,
   ignore: Arc<GitignoreStyleExcludes>,
   executor: task_executor::Executor,
+  symlink_aware: bool,
 }
 
 impl PosixFS {
   pub fn new<P: AsRef<Path>>(
     root: P,
     ignore_patterns: &[String],
     executor: task_executor::Executor,
+  ) -> Result<PosixFS, String> {
+    Self::new_symlink_aware(root, ignore_patterns, executor, true)
+  }
+
+  pub fn new_symlink_aware<P: AsRef<Path>>(
```

Comment:
nit: This can be `symlink-unaware`, meaning that it can ignore symlinks. Maybe change the name to `new_symlink_conscious`, or `new_maybe_symlink_aware`?

```suggestion
  pub fn new_maybe_symlink_aware<P: AsRef<Path>>(
```

Reply:
also nit:
I think the reason for mentioning `symlink_aware` in the function name is to make up for the lack of precision of the signature.
Could be seen as slightly overkill, but if we added an enum:
```
enum SymlinkBehavior {
    Aware,
    Oblivious
}
```
, the signature could be more meaningful by itself and the name's verbosity would be less important.
Maybe then `new_with_symlink_behavior` would be a good name.

Reply:
Good idea: thanks!